src/wwwroot/controllers/PlanCtrl.js

```diff
@@ -174,6 +176,12 @@
       }      
     }
 
+    function isValidUpgradeablePlan() {
+      if($rootScope.accountLimits.requiresDomainConfiguration!=undefined)
+        vm.upgradePlanErrors.isDomainReady = !$rootScope.accountLimits.requiresDomainConfiguration;
+      var isValid=vm.upgradePlanErrors.isDomainReady;
+      return isValid;
```

Comment:
Please, revert this refactoring.

Based on the name, _isValidUpgradeablePlan()_ seems to be a function that is useful to know a particular state: _if the account is in a valid state to allow upgrade the agreement_, right?

But, it also has side effects that are not clear in the function name. In general is not a good idea having functions with side effects.

In this particular case, the problem is that the value of _vm.upgradePlanErrors_ will be only updated if we have executed _isValidUpgradeablePlan()_ before.

Does it make sense?

Reply:
Yes I understand @andresmoschini, I add this behavior to not depend only from backend requests. With this object we have a default value to validate, In the other hand with the other approach we get undefined values until the $rootScope.accountLimits.requiresDomainConfiguration is populated with data from the api.

Reply:
Ok, but a function that seems to simply return a value that has side effects is not acceptable, could you think in another alternative?

Reply:
> With this object we have a default value to validate, In the other hand with the other approach we get undefined values until the $rootScope.accountLimits.requiresDomainConfiguration is populated with data from the api.

That is not a problem, when it is undefined you can decide the default value

Reply:
Perfect @andresmoschini I will change this approach